examples/tv-casting-app/android/App/app/src/main/jni/com/matter/casting/core/MatterCastingPlayer.java

```diff
@@ -162,49 +163,116 @@ public boolean equals(Object o) {
+   *     requested commissioning session.
+   *     <p>For example: To invoke the CastingPlayer/Commissioner-Generated passcode commissioning
+   *     flow, the client would call this API with IdentificationDeclarationOptions containing
+   *     CommissionerPasscode set to true. See IdentificationDeclarationOptions.java for a complete
+   *     list of optional parameters.
+   *     <p>Furthermore, attributes (such as VendorId) describe the TargetApp that the client wants
+   *     to interact with after commissioning. If this value is passed in,
+   *     verifyOrEstablishConnection() will force UDC, in case the desired TargetApp is not found in
+   *     the on-device CastingStore.
+   * @return MatterError - MatterError.NO_ERROR if request submitted successfully, otherwise a
+   *     MatterError object corresponding to the error.
    */
   @Override
   public native MatterError verifyOrEstablishConnection(
-      long commissioningWindowTimeoutSec,
-      EndpointFilter desiredEndpointFilter,
-      MatterCallback<Void> successCallback,
-      MatterCallback<MatterError> failureCallback);
+      ConnectionCallbacks connectionCallbacks,
+      short commissioningWindowTimeoutSec,
+      IdentificationDeclarationOptions idOptions);
 
   /**
-   * Verifies that a connection exists with this CastingPlayer, or triggers a new session request.
-   * If the CastingApp does not have the nodeId and fabricIndex of this CastingPlayer cached on
-   * disk, this will execute the user directed commissioning process.
+   * The simplified version of the verifyOrEstablishConnection() API above.
    *
-   * @return A CompletableFuture that completes when the VerifyOrEstablishConnection is completed.
-   *     The CompletableFuture will be completed with a Void value if the
-   *     VerifyOrEstablishConnection is successful. Otherwise, the CompletableFuture will be
-   *     completed with an Exception. The Exception will be of type
-   *     com.matter.casting.core.CastingException. If the VerifyOrEstablishConnection fails, the
-   *     CastingException will contain the error code and message from the CastingApp.
+   * @param connectionCallbacks contains the onSuccess (Required), onFailure (Required) and
+   *     onCommissionerDeclaration (Optional) callbacks defiend in ConnectCallbacks.java.
+   * @return MatterError - MatterError.NO_ERROR if request submitted successfully, otherwise a
+   *     MatterError object corresponding to the error.
    */
   @Override
-  public MatterError verifyOrEstablishConnection(
-      MatterCallback<Void> successCallback, MatterCallback<MatterError> failureCallback) {
-    return verifyOrEstablishConnection(
-        MIN_CONNECTION_TIMEOUT_SEC, null, successCallback, failureCallback);
+  public MatterError verifyOrEstablishConnection(ConnectionCallbacks connectionCallbacks) {
+    Log.d(TAG, "verifyOrEstablishConnection() overload");
```

Comment:
If you want some logging here, we can be more explicit:

```suggestion
    Log.d(TAG, "verifyOrEstablishConnection()(ConnectionCallbacks)");
```

Reply:
ok